src/hangar/context.py

```diff
@@ -175,9 +175,10 @@ def abort_reader_txn(self, lmdbenv: lmdb.Environment) -> bool:
 class EnvironmentsSingleton(type):
     _instances = {}
     def __call__(cls, *args, **kwargs):
-        if cls not in cls._instances:
-            cls._instances[cls] = super(EnvironmentsSingleton, cls).__call__(*args, **kwargs)
-        return cls._instances[cls]
+        repo_path = kwargs['repo_path']
```

Comment:
What if `repo_path` is not specified during construction?

Reply:
This has been fixed with the validator defined here:

https://github.com/tensorwerk/hangar-py/pull/20/files#diff-0bed44456b6a0cd62fbeaca8f6ab20aeR136

Reply:
However here we assume that `kwargs` contains `repo_path`: I know it's not user-facing, but we need to throw a properly informative exception here.